src/laboratory/static/js/shelfobjectedit.js

```diff
@@ -4,11 +4,35 @@ function processResponseshelfobject(dat){
     $("#closemodal").html(dat["inner-fragments"]["#closemodal"]);
     datatableelement.ajax.reload();
 }
-
-function processResponseshelfobjectDelete(dat) {
-	$('#shelfobjectDelete').html(dat);
-	$("#object_delete").modal('show');
+//Refactored delete method for Shelf Object
+function shelfObjectDelete(obj, shelf_object_id, text) {
+    let url = $(obj).data('url')
+    Swal.fire({ //Confirmation for delete
+        title: 'Are you sure?',
+        confirmButtonText: gettext("Confirm"),
+        showCloseButton: true,
+        denyButtonText: gettext('Cancel'),
+        showDenyButton: true,
+        })
+        .then(function(result) {
+            if (result.isConfirmed) {
+                fetch(url, {
+                    method: "delete",
+                    headers: {'X-CSRFToken': getCookie('csrftoken'), 'Content-Type': 'application/json'},
+                    body: JSON.stringify({'shelfobj': shelf_object_id})})
+                    .then(response => response.json())
+                    .then(data => {
+                        if (data['message']){
+                            Swal.fire(gettext('Success'), data['message'], 'success')
```

Comment:
Something I forgot to add before, after success we want to reload the table since the data changed, the user shouldn't been able to see the object anymore listed, so please do that.
Also, the message should have a timeout like we do on the other place we display success messages, to follow convention.  Example: `Swal.fire({
                icon: 'success',
                title: data.detail,
                showConfirmButton: true,
                timer: 1500
            }); `

Reply:
Added